proxy/proxy.go

```diff
@@ -333,17 +333,17 @@ func (proxy *proxy) init() error {
 	return nil
 }
 
-var nextClientID = uint64(1)
+var nextClientID = uint64(0)
 
 func (proxy *proxy) serveNewClient(proto *protocol, newConn net.Conn) {
+	atomic.AddUint64(&nextClientID, 1)
```

Comment:
Yes, it would make it more clear for other readers.  I'll insert a comment about the potential for a race condition here and squash all of the commits into one.